silver/tests/spec/test_payments.py

```diff
@@ -419,3 +429,208 @@ def test_valid_status_transition_from_unpaid_to_paid(self):
+        self._assert_get_data(random_url, [])
+
+    def test_filters_min_max_amount(self):
+        customer = CustomerFactory()
+
+        payment_below_min = PaymentFactory.create(
+            customer=customer,
+            amount=9.99)
+        payment_min = PaymentFactory.create(
+            customer=customer,
+            amount=10.00)
+        payment_above_min = PaymentFactory.create(
+            customer=customer,
+            amount=10.01)
+        payment_center = PaymentFactory.create(
+            customer=customer,
+            amount=60)
+        payment_below_max = PaymentFactory.create(
+            customer=customer,
+            amount=99.99)
+        payment_max = PaymentFactory.create(
+            customer=customer,
+            amount=100.00)
+        payment_above_max = PaymentFactory.create(
+            customer=customer,
+            amount=100.01)
+
+        url = reverse(
+            'payment-list', kwargs={'customer_pk': customer.pk}
+        )
+
+        range_url = url + '?min_amount=10&max_amount=100'
+        range_url_all = url + '?min_amount=9.99&max_amount=100.01'
+        exact_1_url = url + '?min_amount=60&max_amount=60'
+        no_intersection = url + '?min_amount=100&max_amount=0'
+
+        self._assert_get_data(range_url, [payment_min,
+                                          payment_above_min,
+                                          payment_center,
+                                          payment_below_max,
+                                          payment_max])
+        self._assert_get_data(range_url_all, [payment_below_min,
+                                              payment_min,
+                                              payment_above_min,
+                                              payment_center,
+                                              payment_below_max,
+                                              payment_max,
+                                              payment_above_max])
+        self._assert_get_data(exact_1_url, [payment_center])
+        self._assert_get_data(no_intersection, [])
+
+    def test_filters_currency(self):
+        customer = CustomerFactory()
+
+        payment_usd = PaymentFactory.create(
+            customer=customer,
+            currency='USD')
+        payment_std = PaymentFactory.create(
+            customer=customer,
+            currency='STD')
+
+        url = reverse(
+            'payment-list', kwargs={'customer_pk': customer.pk}
+        )
+
+        usd_iexact_url = url + '?currency=UsD'
+        usd_exact_url = url + '?currency=USD'
+        std_url = url + '?currency=STD'
+        random_url = url + '?currency=RANDOM'
+
+        self._assert_get_data(usd_iexact_url, [payment_usd])
+        self._assert_get_data(usd_exact_url, [payment_usd])
+        self._assert_get_data(std_url, [payment_std])
+        self._assert_get_data(random_url, [])
+
+    def _assert_get_data(self, url, expected_data):
```

Comment:
Different serializer for different test cases. Maybe this could be a thing but we will forget it exists / how would you inject the serializer , as a class param?